samples/subsys/usb/webusb/src/msosv2.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2016-2019 Intel Corporation
+ * Copyright (c) 2023-2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MSOSV2_DESCRIPTOR_H
+#define ZEPHYR_INCLUDE_MSOSV2_DESCRIPTOR_H
+
+/*
+ * Microsoft OS 2.0 platform capability and Microsoft OS 2.0 descriptor set.
+ * See Microsoft OS 2.0 Descriptors Specification for reference.
+ */
+
+#define SAMPLE_MSOS2_VENDOR_CODE	0x02U
+/* Windows version (10)*/
+#define SAMPLE_MSOS2_OS_VERSION		0x0A000000UL
+
+/* random GUID {FA611CC3-7057-42EE-9D82-4919639562B3} */
+#define WEBUSB_DEVICE_INTERFACE_GUID \
+	'{', 0x00, 'F', 0x00, 'A', 0x00, '6', 0x00, '1', 0x00, '1', 0x00, \
+	'C', 0x00, 'C', 0x00, '3', 0x00, '-', 0x00, '7', 0x00, '0', 0x00, \
+	'5', 0x00, '7', 0x00, '-', 0x00, '4', 0x00, '2', 0x00, 'E', 0x00, \
+	'E', 0x00, '-', 0x00, '9', 0x00, 'D', 0x00, '8', 0x00, '2', 0x00, \
+	'-', 0x00, '4', 0x00, '9', 0x00, '1', 0x00, '9', 0x00, '6', 0x00, \
+	'3', 0x00, '9', 0x00, '5', 0x00, '6', 0x00, '2', 0x00, 'B', 0x00, \
+	'3', 0x00, '}', 0x00, 0x00, 0x00, 0x00, 0x00
+
+struct msosv2_descriptor {
```

Comment:
Would it be possible to somehow take the advantage of the new stack to enable automatic `struct msosv2_function_subset_header` insertion when there is more than 1 interface?

The old stack did have https://github.com/zephyrproject-rtos/zephyr/pull/66449 as a solution, but that's fragile approach.

Reply:
I do not think we can handle it. IMO it is better to write a comprehensive documentation about it, with some examples. I think the average user has no idea where and how to use MSOSv2 descriptors.

Reply:
Comprehensive documentation is not a real solution but a way to avoid having to come up with a real solution. While the average user might not have idea about individual MSOSv2 descriptors, it would be good to provide "just-working" experience (if possible).